### ISSUE TITLE
commentedOutCode has many false negatives

### ISSUE THREAD

**Issue by rittneje** [go-critic/go-critic#1457]
```go
package linttest

type Foo struct {
	A string
}

func Bar() {
	// fmt.Println()

	// Baz()

	// x := 1

	f := &Foo{
		// A: "abc",
	}
	_ = f
}

func Baz() {

}
```

All four commented lines should be flagged by the commentedOutCode check. However, only `// fmt.Println()` is.

I am running go-critic via golangci-lint v1.64.5.

**Comment by quasilyte** [comment]
I would guess it's due to the min length check. We should try harder: even if the line is very short, try a couple of guesses to turn a "no" into "yes" without causing new false positives.

I don't remember the exact reasons this length limit exists. Probably because there is not enough checks to cover false-positive cases like:

1. Technically a valid Go code, an expr stmt with identifier. Should not be triggered
```
// foo
```

2. Parsed as a label
```
// example:
```

3. Parsed as a func call with argument
```
// fastpath (usually)
```

There are probably more. We can't catch all cases without introducing some false-positives, but any good heuristics are welcome. The length check usually filters out cases like these just to be safe as we can't competently distinguish them.
